pkg/proxy/winkernel/proxier.go

```diff
@@ -0,0 +1,1105 @@
+	hnsID      string
+	refCount   uint16
+}
+
+func newEndpointInfo(ip string, port uint16, isLocal bool) *endpointsInfo {
+	info := &endpointsInfo{
+		ip:         ip,
+		port:       port,
+		isLocal:    isLocal,
+		macAddress: "00:11:22:33:44:55", // Hardcoding to some Random Mac
+		refCount:   0,
+		hnsID:      "",
+	}
+
+	return info
+}
+
+func (ep *endpointsInfo) Cleanup() {
+	Log(ep, "Endpoint Cleanup", 3)
+	ep.refCount--
+	// Remove the remote hns endpoint, if no service is referring it
+	// Never delete a Local Endpoint. Local Endpoints are already created by other entities.
+	// Remove only remote endpoints created by this service
+	if ep.refCount <= 0 && !ep.isLocal {
+		glog.V(4).Infof("Removing endpoints for %v, since no one is referencing it", ep)
+		deleteHnsEndpoint(ep.hnsID)
+		ep.hnsID = ""
+	}
+
+}
+
+// returns a new serviceInfo struct
+func newServiceInfo(svcPortName proxy.ServicePortName, port *api.ServicePort, service *api.Service) *serviceInfo {
+	onlyNodeLocalEndpoints := false
+	if utilfeature.DefaultFeatureGate.Enabled(features.ExternalTrafficLocalOnly) &&
+		apiservice.RequestsOnlyLocalTraffic(service) {
+		onlyNodeLocalEndpoints = true
+	}
+
+	info := &serviceInfo{
+		clusterIP:  net.ParseIP(service.Spec.ClusterIP),
+		port:       int(port.Port),
+		protocol:   port.Protocol,
+		nodePort:   int(port.NodePort),
+		targetPort: port.TargetPort.IntValue(),
+		// Deep-copy in case the service instance changes
+		loadBalancerStatus:       *helper.LoadBalancerStatusDeepCopy(&service.Status.LoadBalancer),
+		sessionAffinityType:      service.Spec.SessionAffinity,
+		stickyMaxAgeMinutes:      180, // TODO: paramaterize this in the API.
+		loadBalancerSourceRanges: make([]string, len(service.Spec.LoadBalancerSourceRanges)),
+		onlyNodeLocalEndpoints:   onlyNodeLocalEndpoints,
+	}
+
+	copy(info.loadBalancerSourceRanges, service.Spec.LoadBalancerSourceRanges)
+	for _, eip := range service.Spec.ExternalIPs {
+		info.externalIPs = append(info.externalIPs, &externalIPInfo{ip: eip})
+	}
+	for _, ingress := range service.Status.LoadBalancer.Ingress {
+		info.loadBalancerIngressIPs = append(info.loadBalancerIngressIPs, &loadBalancerIngressInfo{ip: ingress.IP})
+	}
+
+	if apiservice.NeedsHealthCheck(service) {
+		p := service.Spec.HealthCheckNodePort
+		if p == 0 {
+			glog.Errorf("Service %q has no healthcheck nodeport", svcPortName.NamespacedName.String())
+		} else {
+			info.healthCheckNodePort = int(p)
+		}
+	}
+
+	return info
+}
+
+type endpointsChange struct {
+	previous proxyEndpointsMap
+	current  proxyEndpointsMap
+}
+
+type endpointsChangeMap struct {
+	lock     sync.Mutex
+	hostname string
+	items    map[types.NamespacedName]*endpointsChange
+}
+
+type serviceChange struct {
+	previous proxyServiceMap
+	current  proxyServiceMap
+}
+
+type serviceChangeMap struct {
+	lock  sync.Mutex
+	items map[types.NamespacedName]*serviceChange
+}
+
+type updateEndpointMapResult struct {
+	hcEndpoints       map[types.NamespacedName]int
```

Comment:
It doesn't look like hcEndpoints is used anywhere in this proxier.  Perhaps just remove the healthcheck stuff if Windows doesn't use it?

Reply:
I actually missed out the code. Will add it back to use healthcheck